File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import Counter

def solution(participant, completion):
    # 각 선수의 참여 횟수를 세기
    participant_counter = Counter(participant)
    completion_counter = Counter(completion)

    # 참여자 배열에서 완주자 배열의 참여 횟수를 빼서 0이 아닌 선수 찾기
    answer = participant_counter - completion_counter

    # 0이 아닌 선수의 이름을 반환
    return list(answer.keys())[0]

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String, Integer> hm = new HashMap<>();
        for(String runner : participant) hm.put(runner, hm.getOrDefault(runner,0) + 1);
        for(String runner : completion) hm.put(runner, hm.get(runner) -1);
        for(String key : hm.keySet()) {
            if(hm.get(key) != 0) {
                answer = key;
                break;
            }
        }
        
        return answer;
    }
}